docs/readmes/proposals/p024_magma_settlement_service.md

```diff
@@ -0,0 +1,281 @@
+As a result of this effort, any MNO, MVNO or 3rd party Gateway Manufacturer will be able to use the software to operate their own roaming interconnect with a network roaming into Helium.
+Out of Scope: The commercial aspects of how an MNO/MVNO purchases data credits on the Helium network through a session purchaser operator is outside the scope of this document and is a commercial arrangement between the clearing house, session purchaser operator and the MNO.
+
+## 2.0 Background
+
+The accounting and settlement of a Helium cellular network is unlike traditional cellular settlement. Unlike traditional cellular networks, a Helium cellular network contains radio and accounting entities (e.g. Miners) owned by private parties not affiliated with the core network operator and/or the subscriber home network provider. Additionally, the subscriber home network operator in a Helium network needs a reward path to pay individual radio/miner owners for the data consumed during subscriber sessions that traverse their deployment. A trust-limited accounting system must be implemented to ensure accounting data integrity and secure the reward path.
+When creating this accounting system, the goal has been to avoid significant (or any) deviations from traditional 3GPP interfaces to allow maximum portability and lower the barriers to entry for vendors.
+The systems centers around the new usage of a standard component, the Online Charging System (OCS). The OCS traditional role is quota management and realtime session management and accounting. In this system, the signaling of standard OCS is repurposed to provide control and accounting functions to a new Helium entity called the Session Purchaser.
+The Session Purchasers job is to account and enforce sessions against a subscribers home network operator account on the helium network. The implementation below describes the role and signaling associated with a Session Purchaser.
+
+## 3.0 Implementation scope
+
+### 3.1 High - level solution architecture
+
+![arch](https://user-images.githubusercontent.com/93994458/149513109-47b80417-7488-49c4-a2f0-ff94e6c57ebf.png)
+
+### 3.2 End to end scenarios
+
+Below will be described e2e scenarios needs to be implemented in scope of this project (Note: scenarios may be decomposed into different use-cases depends on actors involved and/or business case):
+
+- State channel management procedure
+
+The procedure describes Opening, Update and Closure of State channel management call flow.
+The detailed call flow for the procedure is specified in section 3.2.1 “State channel management procedure”.
+
+- UE attach procedure
+
+The procedure in which the UE registers to the network, and receives the usage quota to create the EPS Bearer between the UE and the PGW, in order to be able to send and receive data.
+The detailed call flow for the procedure is specified in section 3.2.2 “UE attach procedure”.
```

Comment:
Note to track a session with OCS, sessiond needs a RG.  Tracking of sessions are not based just on IMSI, but on each RG

On regular LTE, PCRF stores the rules allowed for each susbscriber and provides the RG.  Then that IMSI and RG are used to request and report traffic to OCS

Session Purchaser has to have the concept of RG in order to talk to sessiond